src/linker/Linker.Steps/TypeMapStep.cs

```diff
@@ -137,6 +158,53 @@ static MethodDefinition GetBaseMethodInTypeHierarchy (TypeDefinition type, Metho
 			return null;
 		}
 
+		// Returns a list of default implementations of the given interface method on this type.
+		// Note that this returns a list to potentially cover the diamond case (more than one
+		// most specific implementation of the given interface methods). Linker needs to preserve
+		// all the implementations so that the proper exception can be thrown at runtime.
+		static IEnumerable<InterfaceImplementation> GetDefaultInterfaceImplementations (TypeDefinition type, MethodDefinition interfaceMethod)
+		{
+			// Go over all interfaces, trying to find a method that is an explicit MethodImpl of the
+			// interface method in question.
+			foreach (var interfaceImpl in type.Interfaces) {
+				var potentialImplInterface = interfaceImpl.InterfaceType.Resolve ();
+				if (potentialImplInterface == null)
+					continue;
+
+				bool foundImpl = false;
+
+				foreach (var potentialImplMethod in potentialImplInterface.Methods) {
+					if (potentialImplMethod == interfaceMethod &&
+						!potentialImplMethod.IsAbstract) {
+						yield return interfaceImpl;
+					}
```

Comment:
Should this also break out of the loop? If we've found the exact method match with a body... how could there be another method on the interface which overrides the method we're looking for.

Reply:
In IL you actually can put an override record that overrides a method within the the same class. I don't know what it's good for, but it works and there's tests for that in the CoreCLR tree.

We don't allow this for interfaces but not as a conscious choice, but because it's a spec/implementation deficiency (the runtime will crash with `COR_E_EXECUTIONENGINE` if you do that on interfaces; I just tried).

I don't want to burn into linker more handling for this than the runtime has. If the runtime ever decides it wants to do the same thing for interfaces as it can do for classes, bailing here early would be incorrect.